api/src/main/java/net/kyori/adventure/text/ComponentCompaction.java

```diff
@@ -166,6 +190,41 @@ static Component compact(final @NotNull Component self, final @Nullable Style pa
     return builder.build();
   }
 
+  /**
+  * Checks whether the Component is blank (a TextComponent containing only space characters)
+  *
+  * @param component the component to check
+  * @return true if the provided component is blank, false otherwise
+  */
+  private static boolean isBlank(Component component) {
+    if (component instanceof TextComponent) {
+      final TextComponent textComponent = (TextComponent) component;
+      
+      for (char c : textComponent.content().toCharArray()) {
+        if (c != ' ') return false;
+      }
+      
+      return true;
+    }
+    return false;
+  }
+  
+  /**
+  * Simplify the provided style to remove any information that is redundant,
+  * given that the content is blank
+  *
+  * @param style style to simplify
+  * @return a new, simplified style
+  */
+  private static @NotNull Style simplifyStyleForBlank(final @NotNull Style style) {
+    final Style.Builder builder = style.toBuilder();
+    
+    builder.color(null);
+    builder.decoration(TextDecoration.ITALIC, TextDecoration.State.NOT_SET);
```

Comment:
only italics here?

Reply:
Sorry, I was writing the description. I believe italics are the only redundant styling. If you have others please let me know

Reply:
Ah hm, thinking about it more, makes sense. Could you add a comment in the source here for the other decorations + style attributes and why they're not needed, for the benefit of those reading the code later on?

Reply:
See 1fcc46c